tests/benchdnn/pool/pool.cpp

```diff
@@ -162,15 +162,15 @@ void skip_invalid_prb(const prb_t *prb, res_t *res) {
 // Special function to handle Nvidia libraries issues showing up through the
 // timeline. Not recommended to remove instances to keep working state of any
 // cuda/cuDNN/cuBLAS versions.
-bool cuda_check_correctness(const prb_t *prb,
+bool gpu_check_correctness(const prb_t *prb,
```

Comment:
I'm not quite understanding this flip for all GPUs. The last condition that is not affected by the cuDNN check looks like a cuDNN bug. Why the intention is to let Intel and AMD GPUs go there?

Reply:
There is just one case that fails on Intel/AMD GPUs, below is the case:

```bash
[3214][SRC][1:2:0:3] exp_f32:          64 exp:          64 got:          76 diff:      12 rdiff:  0.1875
[COMPARE_STATS][SRC]: trh=0.078125 err_max_diff:      12 err_max_rdiff:  0.1875 all_max_diff:      28 all_max_rdiff:  0.1875
3627:FAILED (errors:1 total:5746) __REPRO: --pool --engine=gpu --dir=BWD_D --dt=bf16:bf16 ic17ih13oh37kh17ph20
```
The previous specific case works in nVidia bc the condition. The Intel/AMD cases also work if the same condition is applied. That is why this portion of the code is added.

Reply:
If it fails in a generic pooling kernel, it's a kernel issue not accumulating results in f32 as it should.